GVFS/GVFS.Platform.POSIX/POSIXPlatform.cs

```diff
@@ -176,6 +176,11 @@ public override bool IsGitStatusCacheSupported()
             return false;
         }
 
+        public override string GetProgramLocater()

```

Comment:
Feels like the `ProgramLocator` should instead be a type, rather than a string name of a program to execute.

```csharp
interface IExecutableLocator
{
    /// <summary>
    /// Get the full path to the given executable, or return `null` if not found.
    /// </summary>
    string Locate(string executableName);

// or if we want to use the Try* pattern...

    /// <summary>
    /// Try and get the full path to the given executable
    /// </summary>
    bool TryLocate(string executableName, out string fullPath);
}

class (Posix|Windows)ExecutableLocator : IExecutableLocator {...}
```

The implementation on Windows might shell out to `where` and on POSIX platforms to `which`, and might also augment the search paths, or call `which -a` to get the full list, and prefer a particular version over another.

Reply:
Noted for Followup